src/test/java/com/aws/greengrass/logmanager/CloudWatchAttemptLogsProcessorTest.java

```diff
@@ -543,6 +534,91 @@ void GIVEN_unstructured_log_WHEN_breaches_event_size_and_batch_size_limits_THEN_
     }
 
     @Test
+    void GIVEN_component_multiline_pattern_default_WHEN_logs_start_with_whitespace_THEN_append_to_previous_log() throws IOException{
```

Comment:
do any of these tests cause the stack overflow? We need a regression test to make sure it is handled correctly

Reply:
added a test